components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java

```diff
@@ -206,17 +246,180 @@ public void testHandleClaimsForOAuthAuthzReqMessageContext() throws Exception {
+
+        assertEquals(jwtClaimsSet.getAllClaims().size(), 8, "Claims are not successfully set.");
+
+    }
+
+    @Test
+    public void testCustomClaimForOAuthTokenReqMessageContextWithNullAssertion() throws Exception {
+        jwtClaimsSet = new JWTClaimsSet();
+        requestMsgCtx = mock(OAuthTokenReqMessageContext.class);
+        String[] approvedScopes = {"openid", "testScope1", "testScope2"};
+
+        when(requestMsgCtx.getScope()).thenReturn(approvedScopes);
+        when(requestMsgCtx.getProperty(OAuthConstants.OAUTH_SAML2_ASSERTION)).thenReturn(null);
+        when(requestMsgCtx.getProperty(OAuthConstants.ACCESS_TOKEN)).thenReturn(SAMPLE_ACCESS_TOKEN);
+
+        mockStatic(AuthorizationGrantCache.class);
+        authorizationGrantCache = mock(AuthorizationGrantCache.class);
+        AuthorizationGrantCacheEntry authorizationGrantCacheEntry = mock(AuthorizationGrantCacheEntry.class);
+
+        when(AuthorizationGrantCache.getInstance()).thenReturn(authorizationGrantCache);
+        when(authorizationGrantCache.getValueFromCache(any(AuthorizationGrantCacheKey.class))).
+                thenReturn(authorizationGrantCacheEntry);
+
+        AuthenticatedUser user = mock(AuthenticatedUser.class);
+        when(requestMsgCtx.getAuthorizedUser()).thenReturn(user);
+        when(user.isFederatedUser()).thenReturn(false);
+
+        OAuth2AccessTokenReqDTO oAuth2AccessTokenReqDTO = mock(OAuth2AccessTokenReqDTO.class);
+        when(requestMsgCtx.getOauth2AccessTokenReqDTO()).thenReturn(oAuth2AccessTokenReqDTO);
+        when(oAuth2AccessTokenReqDTO.getTenantDomain()).thenReturn(SAMPLE_TENANT_DOMAIN);
+        when(oAuth2AccessTokenReqDTO.getClientId()).thenReturn(SAMPLE_CLIENT_ID);
+
+        mockStatic(OAuth2ServiceComponentHolder.class);
+        applicationManagementService = mock(ApplicationManagementService.class);
+        when(OAuth2ServiceComponentHolder.getApplicationMgtService()).thenReturn(applicationManagementService);
+        when(applicationManagementService.getServiceProviderNameByClientId(anyString(), anyString(), anyString()))
+                .thenReturn(SAMPLE_SERVICE_PROVIDER);
+
+        startTenantFlowProcess();
+
+        samlAssertionClaimsCallback.handleCustomClaims(jwtClaimsSet, requestMsgCtx);
+        assertEquals(jwtClaimsSet.getAllClaims().size(), 8, "Claims are not successfully set.");
+    }
+
+    @Test
+    public void testHandleClaimsForOAuthAuthzReqMessageContextNullAccessToken() throws Exception {
+        jwtClaimsSet = new JWTClaimsSet();
+        oAuthAuthzReqMessageContext = mock(OAuthAuthzReqMessageContext.class);
+        String[] approvedScopes = {"openid", "testScope1", "testScope2"};
+        when(oAuthAuthzReqMessageContext.getApprovedScope()).thenReturn(approvedScopes);
+        when(oAuthAuthzReqMessageContext.getProperty(OAuthConstants.ACCESS_TOKEN))
+                .thenReturn(null);
+
+        OAuth2AuthorizeReqDTO oAuth2AuthorizeReqDTO = mock(OAuth2AuthorizeReqDTO.class);
```

Comment:
Why do you want to mock this. ? Cant you create a new instance of this and use it ?